plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -286,6 +292,100 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static <T extends Type> TypeCoercer<DecimalType, T> createDecimalToInteger(DecimalType fromType, T toType)
+    {
+        if (fromType.isShort()) {
+            return new ShortDecimalToIntegerCoercer<>(fromType, toType);
+        }
+        return new LongDecimalToIntegerCoercer<T>(fromType, toType);
+    }
+
+    private abstract static class AbstractDecimalToIntegerCoercer<T extends Type>
```

Comment:
Can we make the class name similar to the one we use for `AbstractDecimalToIntegerNumberCoercer` ?